### PR TITLE
ENH: add current_session property to BarData

### PR DESCRIPTION
**Pull request by jbredeche**

### CODE REVIEW

File: zipline/_protocol.pyx [hunk 1]
```diff
@@ -747,6 +747,12 @@ cdef class BarData:
         def __set__(self, val):
             self._adjust_minutes = val
 
+    property current_session:
+        def __get__(self):
+            return self._trading_calendar.minute_to_session_label(
```
Comment (ssanderson) [inline zipline/_protocol.pyx]: Can we explicitly pass `direction='next'` here to clarify the behavior in `before_trading_start`?

File: tests/test_bar_data.py [hunk 2]
```diff
@@ -204,6 +204,18 @@ def init_class_fixtures(cls):
 
         cls.ASSETS = [cls.ASSET1, cls.ASSET2]
 
+    def test_current_session(self):
+        for minute in self.trading_calendar.minutes_for_sessions_in_range(
```
Comment (ssanderson) [inline tests/test_bar_data.py]: We should verify the behavior outside of the session as well for BTS.  I think the right behavior there is to roll forward.

File: zipline/_protocol.pyx (hunk 1 above)
Comment (jbredeche) [inline zipline/_protocol.pyx]: yup, done

File: tests/test_bar_data.py (hunk 2 above)
Comment (jbredeche) [inline tests/test_bar_data.py]: agreed, added.

**Pull request merged**